lib/src/connection.dart

```diff
@@ -495,3 +515,66 @@ class Notification {
   /// An optional data payload accompanying this notification.
   final String payload;
 }
+
+class PostgreSQLConnectionPool {
+  PostgreSQLConnectionPool(this.size, this.host, this.port, this.databaseName,
+      {this.username: null,
+        this.password: null,
+        this.timeoutInSeconds: 30,
+        this.timeZone: "UTC",
+        this.useSSL: false});
+
+  final int size;
+  final String host;
+  final int port;
+  final String databaseName;
+  final String username;
+  final String password;
+  final int timeoutInSeconds;
+  final String timeZone;
+  final bool useSSL;
+
+  var _connections = new Map<PostgreSQLConnection, StreamSubscription>();
+
+  PostgreSQLConnection get connection {
+    if(_connections.length == 0) {
+      throw new PostgreSQLException(
+          "Attempting to get connection, but pool is not open.");
+    }
+    PostgreSQLConnection selectedConnection = _connections.keys.first;
+    int maxLengthQueryQueue() => selectedConnection._queryQueue.length;
+    for(var connection in _connections.keys.skip(1))
+      if(connection._queryQueue.length < maxLengthQueryQueue()) {
+        selectedConnection = connection;
+      }
+      return selectedConnection;
+  }
+
+  Future open() async {
+    for(int i = 0; i < size; i++) {
+      _createConnection();
+    }
+    await Future.wait(_connections.keys.map((connection) => connection.open()));
+  }
+
+  void _createConnection()
+  {
+    var connection =new PostgreSQLConnection(host, port, databaseName,
+        username: username,
+        password: password,
+        timeoutInSeconds: timeoutInSeconds,
+        timeZone: timeZone,
+        useSSL: useSSL);
+    StreamSubscription subscription = connection.done.asStream().listen((_){
```

Comment:
Likewise, do we even really need to keep the subscription if that is the case? Can remove the entire Map and just keep a list of connections instead.